localstack/services/lambda_/event_source_listeners/dynamodb_event_source_listener.py

```diff
@@ -76,3 +73,14 @@ def _get_first_and_last_arrival_time(self, first_record, last_record):
             last_record.get("ApproximateArrivalTimestamp", datetime.datetime.utcnow()).isoformat()
             + "Z",
         )
+
+    def _transform_records(self, raw_records: list[dict]) -> list[dict]:
+        """Convert dynamodb.ApproximateCreationDateTime datetime to float"""
+        records_new = []
+        for record in raw_records:
+            record_new = record.copy()
+            if creation_time := record.get("dynamodb", {}).get("ApproximateCreationDateTime"):
+                # convert datetime object to float timestamp
+                record_new["dynamodb"]["ApproximateCreationDateTime"] = creation_time.timestamp()
+            records_new.append(record_new)
+        return records_new
```

Comment:
How is this change related, and why do we need to copy the records instead of modifying in place?

Reply:
It's a parity issue captured by the improved parity tests and is related because it caused parsing issues in the event-ruler.

That's why it's needed: `json.dumps` is not happy with this one ;)
> Convert dynamodb.ApproximateCreationDateTime datetime to float